### PR TITLE
Et3 response save documents

### PR DESCRIPTION
**Pull request by Brogie**
### Change description ###
Added function to save the documents submitted in the ET3 Response form onto the document tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 1]
```diff
@@ -65,4 +70,84 @@ public static List<String> validateEmploymentDates(CaseData caseData) {
...
+     * Adds a document to the ET3 response collection so that it can be displayed on the document tab.
+     * @param caseData data for the current case.
+     * @param documentToAdd the document to be added to the document tab.
+     */
+    public static void addDocument(CaseData caseData, UploadedDocumentType documentToAdd) {
+        if (documentToAdd == null) {
+            return;
+        }
+
+        if (caseData.getEt3ResponseDocumentCollection() == null) {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:104]: CollectionUtils.isEmpty

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 2]
```diff
@@ -65,4 +70,84 @@ public static List<String> validateEmploymentDates(CaseData caseData) {
...
+
+    /**
+     * Checks if a document collection contains a document based on the URL of the document.
+     * @param collectionToCheck a document collection to check.
+     * @param documentUrlToFind the url of the document to be found.
+     * @return true if found, false if not found or the collection passed is null.
+     */
+    private static boolean documentExistsOnCollection(List<DocumentTypeItem> collectionToCheck,
+                                                      String documentUrlToFind) {
+        if (collectionToCheck == null) {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:140]: CollectionUtils.isEmpty

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/Et3ResponseController.java [hunk 3]
```diff
@@ -115,10 +118,45 @@ public ResponseEntity<CCDCallbackResponse> midEmploymentDates(
...
+    @Operation(summary = "display the next steps after ET3 response")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Accessed successfully",
+            content = {
+                @Content(mediaType = "application/json", schema = @Schema(implementation = CCDCallbackResponse.class))
+            }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> aboutToSubmit(
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/Et3ResponseController.java:138]: controller tests

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 4]
```diff
@@ -65,4 +70,84 @@ public static List<String> validateEmploymentDates(CaseData caseData) {
...
+        }
+
+        DocumentType documentType = new DocumentType();
+        documentType.setUploadedDocument(documentToAdd);
+
+        saveFileToCollection(caseData, documentType);
+    }
+
+    /**
+     * Saves a document of DocumentType to the ET3Response document collection.
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:120]: `of DocumentType` feels superfluous since its given by the method signature

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 5]
```diff
@@ -65,4 +70,84 @@ public static List<String> validateEmploymentDates(CaseData caseData) {
...
+        caseData.getEt3ResponseDocumentCollection().add(documentTypeItem);
+    }
+
+    /**
+     * Checks if a document collection contains a document based on the URL of the document.
+     * @param collectionToCheck a document collection to check.
+     * @param documentUrlToFind the url of the document to be found.
+     * @return true if found, false if not found or the collection passed is null.
+     */
+    private static boolean documentExistsOnCollection(List<DocumentTypeItem> collectionToCheck,
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:138]: Could this be moved to a more generic place? It's not tied specifically to ET3 stuff

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 6]
```diff
@@ -65,4 +70,84 @@ public static List<String> validateEmploymentDates(CaseData caseData) {
...
+                                                      String documentUrlToFind) {
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:139]: `toCheck` is a bit wordy. Just "documentUrl" and "collection" are fine without the "ToFind" on the end because its superfluous

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelperTest.java [hunk 7]
```diff
@@ -4,26 +4,42 @@
...
 import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static uk.gov.hmcts.ecm.common.model.helper.Constants.ENGLANDWALES_CASE_TYPE_ID;
 
 class Et3ResponseHelperTest {
 
     public static final String START_DATE_MUST_BE_IN_THE_PAST = "Start date must be in the past";
     public static final String END_DATE_MUST_BE_AFTER_THE_START_DATE = "End date must be after the start date";
     private CaseData caseData;
+    private UploadedDocumentType document1ToSave;
```
Comment (JackReevies) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelperTest.java:25]: `toSave` again as above

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java [hunk 8]
```diff
@@ -65,4 +70,84 @@ public static List<String> validateEmploymentDates(CaseData caseData) {
...
+     */
+    public static void addDocument(CaseData caseData, UploadedDocumentType documentToAdd) {
+        if (documentToAdd == null) {
+            return;
+        }
+
+        if (caseData.getEt3ResponseDocumentCollection() == null) {
+            caseData.setEt3ResponseDocumentCollection(new ArrayList<>());
+        } else if (
+            documentExistsOnCollection(caseData.getEt3ResponseDocumentCollection(),
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3ResponseHelper.java:107]: suggestion: extra out `caseData.getEt3ResponseDocumentCollection()` to a local variable, its used twice and would allow the else if to be on a single line

**Review by JackReevies — CHANGES_REQUESTED** [review]
see comments

**Review by JackReevies — APPROVED** [review]
